apps/core/middlewares/with-routes.ts

```diff
@@ -228,6 +229,13 @@ export const withRoutes: MiddlewareFactory = (next) => {
         return NextResponse.rewrite(url);
       }
 
+      case 'RawHtmlPage': {
+        // fast path for raw html pages
+        const url = createRewriteUrl(`/api/raw-page/${node.id}`, request);
```

Comment:
Curious to know what do you mean by fast path?

Reply:
I mean that's it's an API that returns HTML as directly as possible without any further processing, which is why I'm using an API route (fastest way to get from middleware to HTML output)